rhel/resolver.go

```diff
@@ -0,0 +1,56 @@
+package rhel
+
+import (
+	"context"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+var (
+	_ indexer.Resolver = (*Resolver)(nil)
+)
+
+type Resolver struct{}
+
+// Resolve takes a claircore.IndexReport and uses the rpm Files
+// to determine if the package originated from an RPM. If the package
+// was deemed to have been installed via RPM, it isn't included in the
+// final report.
+func (r *Resolver) Resolve(ctx context.Context, ir *claircore.IndexReport, layers []*claircore.Layer) *claircore.IndexReport {
+	for pkgID, pkg := range ir.Packages {
+		isRPMPackage := false
+	envLoop:
+		for _, env := range ir.Environments[pkgID] {
+			if env.RepositoryIDs != nil {
```

Comment:
nit: not needed. The loop would just skipped. Is it possible `env` is `nil` and that should be checked?

Reply:
Not sure I follow, the `envLoop` would be broken but we'd still get to the files loop with the contextual information that there's nothing to do so go around for another loop

Reply:
I don't think there is a difference between 

```
if env.RepositoryIDs != nil {
	for _, rID := range env.RepositoryIDs {
		...
	}
}
```

and just

```
for _, rID := range env.RepositoryIDs {
	...
}
```

since the loop would not happen if `env.RepositoryIDs == nil`

Reply:
Got it, thanks, wasn't reading them as separate comments

Reply:
sorry I wasn't clear. I was referring to the `env.RepositoryIDs != nil` check prior to the loop but I was also wondering if the intention of that check was indeed for `env.RepositoryIDs` instead of `env` since the loop would handle the case where `env.RepositoryIDs == nil`